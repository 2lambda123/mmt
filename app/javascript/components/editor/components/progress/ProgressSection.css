.section-circle {
    font-size: 2.60em; 
    display: block; 
    float: left; 
    margin-right: 10px
}

.section-label {
    cursor: pointer;
    font-size: 16px;
    font-family: sans-serif;
    padding-left: 4px
}

.pass-circle {
    color: rgb(24, 156, 84)
}

.invalid-circle {
    color: rgb(24, 156, 84)
}