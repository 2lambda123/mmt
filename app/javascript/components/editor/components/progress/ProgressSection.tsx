/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/jsx-no-comment-textnodes */
import { observer } from 'mobx-react'
import React, { ReactNode } from 'react'
import _ from 'lodash'
import { FieldInfo } from './FieldInfo'
import { ProgressCircleType } from './ProgressCircleType'
import ProgressField from './ProgressField'
import './ProgressSection.css'
import withRouter from '../withRouter'
import MetadataEditor from '../../MetadataEditor'

type ProgressSectionProps = {
  section: string
  status: ProgressCircleType
  fields: Array<FieldInfo>
  router: RouterType
  editor: MetadataEditor
}

class ProgressSection extends React.Component<ProgressSectionProps, never> {
  progressSectionCircle(type: ProgressCircleType) {
    switch (type) {
      case ProgressCircleType.Pass:
        return (
          <i className="eui-icon eui-check pass-circle" />
        )
      case ProgressCircleType.NotStarted:
      case ProgressCircleType.Error:
      case ProgressCircleType.Invalid:
        return (
          <i className="eui-icon eui-fa-circle-o invalid-circle" />
        )
      default: return (
        <span>Status Unknown</span>
      )
    }
  }
  navigateTo() {
    const {
      section, router, editor
    } = this.props
    const { params } = router
    const { id } = params
    const sectionName = section.replace(/\s/g, '_')
    setTimeout(() => {
      window.location.href = `/${editor.documentType}/${id}/edit/${sectionName}`
    }, 50)
    // navigate(`/${editor.documentType}/${id}/edit/${sectionName}`, { replace: false })
  }
  render() {
    const {
      section, status, fields, editor
    } = this.props
    const rows = fields.map((field: FieldInfo): ReactNode => (
      <span key={JSON.stringify(field)} data-testid={`progress-field__${_.kebabCase(field.name)}_${field.index}`}>
        <ProgressField fieldInfo={field} editor={editor} />
      </span>
    ))
    return (
      <div className="progress-view" key={JSON.stringify(fields)} data-testid={`progress-section__${_.kebabCase(section)}_${_.kebabCase(status)}`}>
        <div className="section-circle">
          {this.progressSectionCircle(status)}
        </div>
        <div>
          <div className="section-label">
            <div
              onClick={() => this.navigateTo()}
              data-testid={`progress-section__${_.kebabCase(section)}_link`}
            >
              {section}
            </div>
          </div>
          <div>
            {rows}
          </div>
        </div>
      </div>
    )
  }
}
export default withRouter(observer(ProgressSection))
