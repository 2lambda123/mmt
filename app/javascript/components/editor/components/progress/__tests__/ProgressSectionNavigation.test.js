import React from 'react'
import {
  render, fireEvent
} from '@testing-library/react'
import {
  BrowserRouter
} from 'react-router-dom'
import ProgressSection from '../ProgressSection'

// Moved this ino a separate file for consistency, but it really should be included
// with ProgressSection.test.js once the comments in ProgressFieldNavigation.test.js
// are resolved.
describe('Progress field navigation test array field', () => {
  test('navigate to edit a field in array', async () => {
    const props = {
      section: 'Tool Contacts',
      status: 'Pass',
      fields: [
        {
          section: 'Tool Contacts',
          name: 'ContactGroups',
          index: null,
          status: 'Not Started',
          required: true
        },
        {
          section: 'Tool Contacts',
          name: 'ContactPersons',
          index: null,
          status: 'Not Started',
          required: false
        }
      ]
    }
    const { queryByTestId, container } = render(
      <BrowserRouter>
        <ProgressSection {...props} />
      </BrowserRouter>
    )
    const spy = jest.spyOn(ProgressSection.WrappedComponent.prototype, 'navigateTo')
    const clickField = queryByTestId('progress-section__tool-contacts_link')
    fireEvent.click(clickField)
    expect(spy).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
  })
})
