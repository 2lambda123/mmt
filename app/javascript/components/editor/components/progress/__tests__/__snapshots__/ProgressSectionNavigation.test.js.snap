// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress field navigation test array field navigate to edit a field in array 1`] = `
<div>
  <div
    data-testid="progress-section__tool-contacts_pass"
    style="display: flex; align-items: center;"
  >
    <div
      class="section-circle"
    >
      <i
        class="eui-icon eui-check pass-circle"
      />
    </div>
    <div>
      <div
        class="section-label"
      >
        <div
          data-testid="progress-section__tool-contacts_link"
        >
          Tool Contacts
        </div>
      </div>
      <div>
        <span
          data-testid="progress-field__contact-groups_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-groups_null_required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-groups_link"
            >
              <i
                class="eui-icon eui-required-o not-started-required-circle"
                title="ContactGroups"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__contact-persons_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-persons_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-persons_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="ContactPersons"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
